### PR TITLE
[FLINK-30880] Update flink version to 1.16.1

### PR DESCRIPTION
**Pull request by eskabetxe**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        uses: actions/setup-java@v2
+        with:
+          java-version: ${{ matrix.jdk }}
+          distribution: 'temurin'
+          cache: 'maven'
+
+      - name: Set Maven 3.8.6
+        uses: stCarolas/setup-maven@v4.2
+        with:
+          maven-version: 3.8.6
```
[the remark this answers is not in the document]
Comment (zentol) [reply .github/workflows/ci.yml:74]: Please dont upgrade maven version beyond 3.8.6 without properly investigating the repercussions of that.

**Review by zentol — CHANGES_REQUESTED** [review]
You dont need a custom ci.yml. Just modify the link passed in push_pr.yml to the shared workflow.

**Comment by eskabetxe** [comment]
@zentol its done. I update the branch with another test erroneous..

The version is updated but it will fail again if a new bugfix is released..
Maybe we should change the full url to archive, knowing that it is very slow..
https://archive.apache.org/dist/flink/flink-1.16.1/flink-1.16.1-bin-scala_2.12.tgz

**Comment by zentol** [comment]
>  Maybe we should change the full url to archive, knowing that it is very slow..

Maybe, yes. The caching in the reusable workflow was broken, but it should work now, so the download speed shouldnt be an issue since it'll only happen once (every few weeks)

**Review by zentol — APPROVED** [review]

**Pull request merged**